ArduCopter/mode_guided.cpp

```diff
@@ -179,22 +178,15 @@ void ModeGuided::posvel_control_start()
     // set guided_mode to velocity controller
     guided_mode = SubMode::PosVel;
 
-    pos_control->init_xy_controller();
-
-    // set speed and acceleration from wpnav's speed and acceleration
-    pos_control->set_max_speed_xy(wp_nav->get_default_speed_xy());
-    pos_control->set_max_accel_xy(wp_nav->get_wp_acceleration());
-
-    const Vector3f& curr_pos = inertial_nav.get_position();
-    const Vector3f& curr_vel = inertial_nav.get_velocity();
+    // initialise horizontal speed, acceleration
+    pos_control->set_max_speed_accel_xy(wp_nav->get_default_speed_xy(), wp_nav->get_wp_acceleration());
 
-    // set target position and velocity to current position and velocity
-    pos_control->set_xy_target(curr_pos.x, curr_pos.y);
-    pos_control->set_desired_velocity_xy(curr_vel.x, curr_vel.y);
+    // initialize vertical speeds and acceleration
+    pos_control->set_max_speed_accel_z(wp_nav->get_default_speed_down(), wp_nav->get_default_speed_up(), wp_nav->get_accel_z());
 
-    // set vertical speed and acceleration
-    pos_control->set_max_speed_z(wp_nav->get_default_speed_down(), wp_nav->get_default_speed_up());
-    pos_control->set_max_accel_z(wp_nav->get_accel_z());
+    // initialise velocity controller
+    pos_control->init_z_controller();
+    pos_control->init_xy_controller();
```

Comment:
I agree. As this is a 2.5D controller I would do it the way we have done here. It will take time to get all the use cases set up properly as there are some structural issues that will need to be changed.